vermouth/processors/do_mapping.py

```diff
@@ -48,25 +48,32 @@ class GraphMapping:
     #       or we need to at least the garantee we run it all in order. Which we
     #       can't unless we do run_system instead of run_molecule. We should
     #       maybe also move this class to a different file, but we'll see.
-    def __init__(self, blocks_from, blocks_to, mapping, extra=()):
+    def __init__(self, blocks_from, blocks_to, mapping, weights=None, extra=()):
         """
         blocks_from and blocks_to are sequences of Blocks.
         Mapping is a dictionary of {(residx, atomname): [(residx, atomname), ...], ...}.
         residx in these cases is the index of the residue in blocks_from and
         blocks_to respectively.
         """
+        if weights is None:
+            weights = {}
         self.block_from = self._merge(blocks_from)
         self.block_to = self._merge(blocks_to)
 
         self.mapping = defaultdict(set)
+        self.weights = defaultdict(dict)
         # Translate atomnames in mapping to node keys.
         for from_, to in mapping.items():
             res_from, name_from = from_
+            from_idxs = list(self.block_from.find_atoms(atomname=name_from, resid=res_from))
             for res_to, name_to in to:
-                from_idxs = self.block_from.find_atoms(atomname=name_from, resid=res_from)
                 to_idxs = self.block_to.find_atoms(atomname=name_to, resid=res_to)
                 for to_idx in to_idxs:
                     self.mapping[to_idx].update(from_idxs)
+                    if from_idxs:
+                        self.weights[to_idx][from_idxs[0]] = (
+                            weights[(res_to, name_to)][(res_from, name_from)]
+                        )
```

Comment:
Why only `from_idxs[0]`? I'm lost in my own code :s

Reply:
I am not completely sure either from the top of my head. But I'd rather have some tests in place before messing around with some refactoring.